.github/workflows/ci.yml

```
@@ -0,0 +1,24 @@
+name: Cargo Build & Test
+
+on:
+  push:
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build_and_test:
+    name: Rust project - latest
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        toolchain:
+          - stable
+          - beta
+          - nightly
+    steps:
+      - uses: actions/checkout@v4
+      - run: rustup update ${{ matrix.toolchain }} && rustup default ${{ matrix.toolchain }}
```

Isn't it sufficient to to build and test on stable?

There's also a more concise way to ensure what a particular Cargo tool is available:

```suggestion
    steps:
      - uses: actions/checkout@v4
      - uses: dtolnay/rust-toolchain@stable
```